src/Proto.Actor/EventStream/DeadLetter.cs

```diff
@@ -46,15 +46,16 @@ public DeadLetterProcess(ActorSystem system) : base(system)
 
         protected internal override void SendUserMessage(PID pid, object message)
         {
+            var (msg, sender, header) = MessageEnvelope.Unwrap(message);
+
             if (System.Metrics.Enabled)
             {
                 ActorMetrics.DeadletterCount.Add(1,
                     new("id", System.Id), new("address", System.Address),
-                    new("messagetype", MessageEnvelope.UnwrapMessage(message)?.GetType().Name ?? "<null>")
+                    new("messagetype", msg)
```

Comment:
This is the raw object, we should still get the type name ;)

Reply:
fixed